### PR TITLE
Offline upgrade v1.12

### LINKED ISSUE #4978: Offline upgrade
## Motivation

We cannot implement dumpless upgrade yet.

So in the meantime, we implement help for migration from 
- v1.10 -> v1.11
- v1.11 -> v1.12

As we did for [v1.9 -> v1.10](https://github.com/meilisearch/meilisearch/releases/tag/v1.9.1)

## Impacted teams

Should help support for @meilisearch/cloud-team

### PR DESCRIPTION
**Pull request by ManyTheFish**
# Pull Request

## Related issue
Fixes #4978 

## What does this PR do?
- add v1_11_to_v1_12 function to upgrade Meilisearch from v1.11 to v1.12
- Convert the update files from OBKV to ndjson format

### CODE REVIEW

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Would it be possible to add much more `.context` uses, like on every `?`, please? The thing is that we are doing a lot of file operations and they never display the file names on errors. I recommend using `with_context(|| format!("while ... on {:?}", file.display))?`.

**Review by Kerollmops — CHANGES_REQUESTED** [review]
It looks much better to me, thank you 🧑‍🍳
However, can you clean up the commits, please?

**Review by Kerollmops — DISMISSED** [review]
Love it. Thank you 👍
bors merge

File: crates/meilitool/src/upgrade/v1_12.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            }
+
+            serde_json::to_writer(&mut file, &json_document)?;
+        }
+
+        let file = file.into_inner().map_err(|e| e.into_error()).context(format!(
+            "while flushing update file bufwriter for update file {update_file_path:?}"
+        ))?;
+        let _ = file
+            .persist(&update_file_path)
```
Comment (dureuill) [inline crates/meilitool/src/upgrade/v1_12.rs:72]:
```suggestion
            // replace the obkv file with the rewritten NDJSON file
            .persist(&update_file_path)
```

**Comment by dureuill** [comment]
I added a clarification comment

**Review by irevoire — APPROVED** [review]
-> The general process seems good to me
-> Arroy didn't change
-> The index-scheduler works with the new batches route :tada: 
Thanks @ManyTheFish :fire: 

bors merge

**Pull request merged**